File: 测试界面/响应链与事件传递/ViewController.m
//
//  ViewController.m
//  响应链与实践传递
//
//  Created by saidicaprio on 2019/5/10.
//  Copyright © 2019 贺嘉炜. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    UIView *view = [[UIView alloc] initWithFrame:self.view.bounds];
    
    [self.view addSubview:view];
    UIGestureRecognizer *rec = [[UIGestureRecognizer alloc] initWithTarget:self action:@selector(onTouch:)];
    UIApplication
    
    // Do any additional setup after loading the view.
}


@end

File: 测试界面/TextField的DidBeginEditing代理/ViewController.swift
//
//  ViewController.swift
//  TextField的DidBeginEditing代理
//
//  Created by 贺嘉炜 on 2017/7/10.
//  Copyright © 2017年 贺嘉炜. All rights reserved.
//

import UIKit

class ViewController: UIViewController ,UITextFieldDelegate {

    @IBOutlet var textField: UITextField!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    func textFieldDidBeginEditing(_ textField: UITextField) {
//        textField.resignFirstResponder()
    }

    func textFieldDidEndEditing(_ textField: UITextField) {
        
    }

}

